sycl/test/esimd/vadd.cpp

```diff
@@ -1,11 +1,20 @@
-// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -o %t-lgcy.out
+// RUN: %RUN_ON_HOST %t-lgcy.out
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -o %t.out
 // RUN: %RUN_ON_HOST %t.out
 
-// Check that the code compiles with -O0 and -g
-// RUN: %clangxx -I %sycl_include %s -o %t.out -fsycl -O0
-// RUN: %clangxx -I %sycl_include %s -o %t.out -fsycl -O0 -g
+// Check that the code compiles with -O0 and -g on both legacy and new Pass
+// Managers
+// RUN: %clangxx -I %sycl_include %s -o %t.out -fsycl -fno-legacy-pass-manager -O0
```

Comment:
why this is added to esimd tests? How these 3 tests were selected for such modification?

Reply:
They crash on SYCLLowerESIMD pass execution step if loop transformation and vectorization passes are enabled in CodeGen.

Reply:
@kbobrovs Could you please provide feedback or approve the PR if there are no more concerns?

Reply:
I see. @mlychkov - could you please instead mark them as XFAIL, add a TODO and file an internal ticket?

Reply:
@kbobrovs, the point is that those tests fail with new pass manager _without_ changes proposed by @mlychkov, but they do pass with this PR using both pass managers

Reply:
Ok, I think I get it now. Thanks!